secret_contracts/coinjoin/Cargo.toml

```diff
@@ -1,5 +1,5 @@
 [package]
-name = "contract"
+name = "coinjoin"
```

Comment:
@reuvenpo: it was my understanding from some prior documentation that the name would always be `contract`. This requirement was coming from `core`. I see now that you are changing this name to a more meaningful name (which makes sense), but this breaks `discovery-cli` (which can be fixed), but I want to understand the implications of this change before I propagate them to the `discovery-cli`. Any comments, please? :)

Cc: @fredfortier

Reply:
I did not properly communicate to Fred that this branch is a WIP, and he merged it assuming that it all works. All i did was check that the code compiles under `cargo check`. I think we should revert this merge until we resolve all these issues and then merge the branch when we're happy with it